Bottom: cd32cbbd25572322b93ecfc284d3b5ec81d146ba
Top:    a5acc3d0b9720a8b2aab72dc838b3105226c7f8e
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-11-13 10:15:12 -0800

mac80211: Add precise time-stamps to association logging.

This helps understand exactly how long it takes to connect
to APs.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 1722ead..9861550 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -2630,7 +2630,14 @@ static void ieee80211_rx_mgmt_auth(struct ieee80211_sub_if_data *sdata,
 
 	event.u.mlme.status = MLME_SUCCESS;
 	drv_event_callback(sdata->local, sdata, &event);
-	sdata_info(sdata, "authenticated\n");
+
+	{
+		struct timeval tv;
+		do_gettimeofday(&tv);
+		sdata_info(sdata, "authenticated at: %lu.%lu\n",
+			   tv.tv_sec, tv.tv_usec);
+	}
+
 	ifmgd->auth_data->done = true;
 	ifmgd->auth_data->timeout = jiffies + IEEE80211_AUTH_WAIT_ASSOC;
 	ifmgd->auth_data->timeout_started = true;
@@ -3134,10 +3141,16 @@ static void ieee80211_rx_mgmt_assoc_resp(struct ieee80211_sub_if_data *sdata,
 	status_code = le16_to_cpu(mgmt->u.assoc_resp.status_code);
 	aid = le16_to_cpu(mgmt->u.assoc_resp.aid);
 
-	sdata_info(sdata,
-		   "RX %sssocResp from %pM (capab=0x%x status=%d aid=%d)\n",
-		   reassoc ? "Rea" : "A", mgmt->sa,
-		   capab_info, status_code, (u16)(aid & ~(BIT(15) | BIT(14))));
+	{
+		struct timeval tv;
+		do_gettimeofday(&tv);
+		sdata_info(sdata,
+			   "RX %sssocResp from %pM (capab=0x%x status=%d aid=%d) at: %lu.%lu\n",
+			   reassoc ? "Rea" : "A", mgmt->sa,
+			   capab_info, status_code,
+			   (u16)(aid & ~(BIT(15) | BIT(14))), tv.tv_sec,
+			   tv.tv_usec);
+	}
 
 	pos = mgmt->u.assoc_resp.variable;
 	ieee802_11_parse_elems(pos, len - (pos - (u8 *) mgmt), false, &elems);
@@ -3176,7 +3189,12 @@ static void ieee80211_rx_mgmt_assoc_resp(struct ieee80211_sub_if_data *sdata,
 		}
 		event.u.mlme.status = MLME_SUCCESS;
 		drv_event_callback(sdata->local, sdata, &event);
-		sdata_info(sdata, "associated\n");
+		{
+			struct timeval tv;
+			do_gettimeofday(&tv);
+			sdata_info(sdata, "associated at: %lu.%lu\n",
+				   tv.tv_sec, tv.tv_usec);
+		}
 
 		/*
 		 * destroy assoc_data afterwards, as otherwise an idle
@@ -3721,9 +3739,14 @@ static int ieee80211_auth(struct ieee80211_sub_if_data *sdata)
 
 	drv_mgd_prepare_tx(local, sdata);
 
-	sdata_info(sdata, "send auth to %pM (try %d/%d)\n",
-		   auth_data->bss->bssid, auth_data->tries,
-		   IEEE80211_AUTH_MAX_TRIES);
+	{
+		struct timeval tv;
+		do_gettimeofday(&tv);
+		sdata_info(sdata, "send auth to %pM (try %d/%d) at: %lu.%lu\n",
+			   auth_data->bss->bssid, auth_data->tries,
+			   IEEE80211_AUTH_MAX_TRIES, tv.tv_sec,
+			   tv.tv_usec);
+	}
 
 	auth_data->expected_transaction = 2;
 
@@ -4538,7 +4561,12 @@ int ieee80211_mgd_auth(struct ieee80211_sub_if_data *sdata,
 					    WLAN_REASON_UNSPECIFIED);
 	}
 
-	sdata_info(sdata, "authenticate with %pM\n", req->bss->bssid);
+	{
+		struct timeval tv;
+		do_gettimeofday(&tv);
+		sdata_info(sdata, "authenticate with %pM at: %lu.%lu\n",
+			   req->bss->bssid, tv.tv_sec, tv.tv_usec);
+	}
 
 	err = ieee80211_prep_connection(sdata, req->bss, false, false);
 	if (err)
